libs/remote_storage/src/azure_blob.rs

```diff
@@ -117,13 +117,22 @@ impl AzureBlobStorage {
     ) -> Result<Download, DownloadError> {
         let mut response = builder.into_stream();
 
+        let mut etag = None;
+        let mut last_modified = None;
         let mut metadata = HashMap::new();
         // TODO give proper streaming response instead of buffering into RAM
         // https://github.com/neondatabase/neon/issues/5563
 
         let mut bufs = Vec::new();
         while let Some(part) = response.next().await {
             let part = part.map_err(to_download_error)?;
+            let etag_str: &str = part.blob.properties.etag.as_ref();
```

Comment:
What a strange API Azure provides here: a stream of byte chunks within a blob, but each chunk specifies the blob's metadata as well.  Oh well!

Reply:
This might be their way to represent large downloads that might span multiple HTTP requests. AWS has the same but exposes a "small blob" API as well. Multiple HTTP requests mean multiple response headers.